crates/telio-firewall/src/firewall.rs

```diff
@@ -1118,6 +1170,101 @@ impl Firewall for StatefullFirewall {
             telio_log_error!("TCP cache poisoned");
         };
 
+        if let Ok(mut udp_conn_cache) = self.udp.lock() {
+            telio_log_debug!(
+                "Inspecting {} UDP connections for reset",
+                udp_conn_cache.len()
+            );
+
+            const IPV4_HEADER_LEN: usize = 20;
+            const ICMP_HEADER_LEN: usize = 8;
+            const ICMP_DATA_LEN: usize = UdpConnectionInfo::LAST_PKG_HEADER;
+
+            let mut ipv4pkgbuf = [0u8; IPV4_HEADER_LEN + ICMP_HEADER_LEN + ICMP_DATA_LEN];
+
+            // Write most of the fields for ICMP and IP packets upfront
+            #[allow(clippy::expect_used)]
+            #[allow(index_access_check)]
+            let mut ipv4pkg = MutableIpv4Packet::new(&mut ipv4pkgbuf[..IPV4_HEADER_LEN])
+                .expect("IPv4 buffer should not be too small");
+            ipv4pkg.set_version(4);
+            ipv4pkg.set_header_length(5);
+            ipv4pkg.set_total_length((IPV4_HEADER_LEN + ICMP_HEADER_LEN + ICMP_DATA_LEN) as _);
+            ipv4pkg.set_flags(Ipv4Flags::DontFragment);
+            ipv4pkg.set_ttl(0xFF);
+            ipv4pkg.set_next_level_protocol(IpNextHeaderProtocols::Icmp);
+            ipv4pkg.set_source(StdIpv4Addr::LOCALHOST);
```

Comment:
This should be the tunnel address instead of localhost to make it clear that meshnet's interface emited the stuff

Reply:
Oh yeah, it's a leftover as the real source setup is 30 lines below. Thanks for pointing this out

Reply:
Done